test/boost/flat_mutation_reader_test.cc

```diff
@@ -728,124 +728,122 @@ SEASTAR_TEST_CASE(test_make_forwardable) {
     });
 }
 
-SEASTAR_TEST_CASE(test_make_nonforwardable) {
-    return seastar::async([] {
-        simple_schema s;
-        tests::reader_concurrency_semaphore_wrapper semaphore;
-        const auto permit = semaphore.make_permit();
+SEASTAR_THREAD_TEST_CASE(test_make_nonforwardable) {
```

Comment:
This should not be a separate commit, instead the change should be retroactively applied to the past commits.